instance/instance.go

```diff
@@ -474,70 +488,89 @@ func (s *Store) UpdateObservations(w http.ResponseWriter, r *http.Request) {
-			if task.State != "" {
-				if task.State != models.CompletedState {
-					validationErrs = append(validationErrs, fmt.Errorf("bad request - invalid task state value: %v", task.State))
-				} else if err := s.UpdateBuildHierarchyTaskState(id, task.DimensionName, task.State); err != nil {
-					log.Error(err, nil)
-					http.Error(w, "Failed to update build hierarchy task state", http.StatusInternalServerError)
-					return
+		if tasks.BuildHierarchyTasks != nil {
+			for _, task := range tasks.BuildHierarchyTasks {
+				if task.State != "" {
+					if task.State != models.CompletedState {
+						validationErrs = append(validationErrs, fmt.Errorf("bad request - invalid task state value: %v", task.State))
+					} else if err := s.UpdateBuildHierarchyTaskState(id, task.DimensionName, task.State); err != nil {
+						audit.LogError(ctx, errors.WithMessage(err, "Failed to update build hierarchy task state"), data)
+						return &updateTaskErr{err, http.StatusInternalServerError}
+					}
 				}
 			}
 		}
-	}
 
-	if tasks.BuildSearchIndexTasks != nil {
-		for _, task := range tasks.BuildSearchIndexTasks {
-			if task.State != "" {
-				if task.State != models.CompletedState {
-					validationErrs = append(validationErrs, fmt.Errorf("bad request - invalid task state value: %v", task.State))
-				} else if err := s.UpdateBuildSearchTaskState(id, task.DimensionName, task.State); err != nil {
-					log.Error(err, nil)
-					http.Error(w, "Failed to update build search index task state", http.StatusInternalServerError)
-					return
+		if tasks.BuildSearchIndexTasks != nil {
+			for _, task := range tasks.BuildSearchIndexTasks {
+				if task.State != "" {
+					if task.State != models.CompletedState {
+						validationErrs = append(validationErrs, fmt.Errorf("bad request - invalid task state value: %v", task.State))
+					} else if err := s.UpdateBuildSearchTaskState(id, task.DimensionName, task.State); err != nil {
+						audit.LogError(ctx, errors.WithMessage(err, "Failed to update build search index task state"), data)
+						return &updateTaskErr{err, http.StatusInternalServerError}
+					}
 				}
 			}
 		}
-	}
 
-	if len(validationErrs) > 0 {
-		for _, err := range validationErrs {
-			log.Error(err, nil)
+		if len(validationErrs) > 0 {
+			for _, err := range validationErrs {
+				audit.LogError(ctx, errors.WithMessage(err, "validation error"), data)
+			}
+			// todo: add all validation errors to the response
+			return &updateTaskErr{validationErrs[0], http.StatusBadRequest}
+		}
+		return nil
+	}()
+
+	if updateErr != nil {
+		if auditErr := s.Auditor.Record(ctx, updateImportTaskAction, audit.Unsuccessful, ap); auditErr != nil {
+			updateErr = &updateTaskErr{errs.ErrInternalServer, http.StatusInternalServerError}
 		}
-		// todo: add all validation errors to the response
-		http.Error(w, validationErrs[0].Error(), http.StatusBadRequest)
+		audit.LogError(ctx, errors.WithMessage(updateErr, "updateImportTask endpoint: request unsuccessful"), data)
```

Comment:
Should be `log.ErrorCtx(...)` and log should take place before calling auditor and the manipulating of updateErr, we need to log out the actual error